dash/dash.py

```diff
@@ -19,18 +22,51 @@
 from .development.base_component import Component
 from . import exceptions
 from ._utils import AttributeDict as _AttributeDict
+from ._utils import interpolate_str as _interpolate
+
+_default_index = '''
+<!DOCTYPE html>
+<html>
+    <head>
+        {metas}
+        <title>{title}</title>
+        {favicon}
+        {css}
+    </head>
+    <body>
+        {app_entry}
+        <footer>
+            {config}
+            {scripts}
```

Comment:
Could we split out scripts into `dash_renderer_scripts` and `dash_components_scripts`? The main reason is that `dash_renderer` is going to become configurable in when we add in custom JS hooks. So, users will end up needing to do something like:
```
dash_renderer = DashRenderer({
    request_hook: function(...) {
         ...
    }
})
```

And so it'd be nice if they could replace this:
```
<!DOCTYPE html>
<html>
    <head>
        {metas}
        <title>{title}</title>
        {favicon}
        {css}
    </head>
    <body>
        {app_entry}
        <footer>
            {config}
            {dash_renderer}
            {dash_component_scripts}
        </footer>
    </body>
</html>
```

with something like this:
```
<!DOCTYPE html>
<html>
    <head>
        {metas}
        <title>{title}</title>
        {favicon}
        {css}
    </head>
    <body>
        {app_entry}
        <footer>
            {config}
            dash_renderer = DashRenderer({
                request_hook: function(...) {
                     ...
                }
            })
            {dash_component_scripts}
        </footer>
    </body>
</html>
```

Now, I suppose the other way they could do this would be with the interpolated index function, where they would do something like:
```
class CustomDash(dash):
    def interpolated_index(metas, title, css, config, scripts, _app_entry, favicon):
        filtered_scripts = [
            script for script in scripts if 'dash_renderer' not in script
        ]

        return '''
        <!DOCTYPE html>
        <html>
            <head>
                {metas}
                <title>{title}</title>
                {favicon}
                {css}
            </head>
            <body>
                {app_entry}
                <footer>
                    {config}
                    dash_renderer = DashRenderer({
                        request_hook: function(...) {
                             ...
                        }
                    })
                    {filtered_scripts}
                </footer>
            </body>
        </html>
        '''.format(metas, title, css, config, filtered_scripts, _app_entry, favicon)
```

Am I understanding that correctly?

Reply:
I wasn't too sure about the js hooks so I left them out of this, right now the scripts are all bundled together in _generate_scripts_html that return a string with all the scripts tags, can't iterate over it but it could change. 

I thought we could have a config to disable the includes of dash-renderer. Then the user can append a custom renderer to the scripts resources or include it a custom index.

Reply:
If they're customizing the `DashRenderer` don't they need to have the tags to load the uncustomized one first?